partcad-client-utils/src/partcad_client_utils/selfupdate.py

```
@@ -0,0 +1,655 @@
+        return
+    with tarfile.open(archive_path, "r:gz") as tf:
+        members = tf.getmembers()
+        _reject_traversal([m.name for m in members], dest)
+        # `filter="data"` is the default from Python 3.14 and rejects unsafe
+        # members outright; passing it explicitly keeps the newer 3.10-3.13
+        # patch releases identical. The older ones have no such parameter at
+        # all, and fall back to the path check made above.
+        try:
+            tf.extractall(dest, members=members, filter="data")  # nosec B202 - members validated above
+        except TypeError:
+            tf.extractall(dest, members=members)  # nosec B202 - members validated above
```

Ruff (0.16.1)

[error] Uses of `tarfile.extractall()`

(S202)